corehq/apps/reports/standard/users/reports.py

```diff
@@ -155,9 +156,9 @@ def _user_history_row(record, domain, timezone):
         cached_user_id_to_username(record.user_id),
         cached_user_id_to_username(record.changed_by),
         _get_action_display(record.action),
-        record.details['changed_via'],
-        record.message,
-        _user_history_details_cell(record.details['changes'], domain),
+        record.changed_via,
+        UserChangeMessageFormatterV1().get_messages(record.change_messages),
```

Comment:
I imagine, when there is a new version, the log record would need to know to the version it was created for. So, UserChangeMessageV2.version and UserChangeFormatterV2.version should both be equal to 2. So when log is created with UserChangeMessageV2 we save "2" as the version on the log and hence use `UserChangeFormatterV2`.
For now there is only one version so we don't need that column as of now and UserChangeFormatterV2.version could be assumed to be `None` or keep it as 1 as set default value for version in log to be 1.

>  I'd expect the formatter to know how to handle possible messages regardless of version.

Formatter would know that based on the syntax of the `record.change_messages`? 
Or `record.change_messages` itself can have a version entry instead of a separate column said above but it would be nice to just have a separate column only so that change_messages are only for messages?

Reply:
Storing a version as part of the log definitely makes sense.  I agree with you that it's probably fine to wait until there's a v2 to add a version column.  I was just thinking that it'd be nice for something in `change_messages.py` to be responsible for checking the version and figuring out how to render, so reports and other callers could just use that without needing to worry about it.

Reply:
> for something in change_messages.py to be responsible for checking the version

Ya that would have been nice. It would mean some logic in the code, and hence branching. Using a column itself would avoid the need to check for version each time the message is to be formatted which sounds like an overkill. 
Thinking about it now, 
We could also pass the record instead of passing the messages from the view, this way, the caller would not need to pass version separately and formatter/layer above formatter could get the version from the record itself keeping things cleaner.

Reply:
the way I see this working would be something like the following:
```
# v1
slug = "set_program"

#v2
slug = "set_program_2"

formatters = {
    "set_program": format_set_program,
    "set_program_2": format_set_program_2
}
```

I don't see the need to version the entire formatter class unless we are making fundamental changes to the message structure.